src/components/Taxonomy/Taxonomy.tsx

```diff
@@ -183,15 +203,14 @@ const Item = ({ item, isFiltering }: { item: TaxonomyItem, isFiltering?: boolean
             disabled={disabled}
             checked={checked}
             ref={setIndeterminate}
-            onChange={e => setSelected(item.path, e.currentTarget.checked)}
+            onChange={e => setSelected(path, e.currentTarget.checked)}
           />
-          {item.label}
+          {name}
         </label>
         {!isFiltering && (
```

Comment:
I see that this menu doesn't hide in search mode, so maybe that's because reconciliation doesn't treat this Item as changed, that's strange...